cogs/TextFilter.py

```diff
@@ -0,0 +1,701 @@
+#!/usr/bin/env python
+
+"""
+Koala Bot Text Filter Code
+"""
+
+# Built-in/Generic Imports
+import os
+import asyncio
+import time
+import re
+import logging
+
+# Libs
+from discord.ext import commands, tasks
+import discord
+
+# Own modules
+import KoalaBot
+from utils.KoalaColours import *
+from utils.KoalaUtils import error_embed, is_channel_in_guild, extract_id
+from utils.KoalaDBManager import KoalaDBManager
+
+
+def text_filter_is_enabled(ctx):
+    """
+    A command used to check if the guild has enabled TextFilter
+    e.g. @commands.check(KoalaBot.is_admin)
+    :param ctx: The context of the message
+    :return: True if admin or test, False otherwise
+    """
+    try:
+        result = KoalaBot.check_guild_has_ext(ctx, "TextFilter")
+    except PermissionError:
+        result = False
+
+    return result or (str(ctx.author) == KoalaBot.TEST_USER and KoalaBot.is_dpytest)
+
+
+class TextFilter(commands.Cog, name="TextFilter"):
+    """
+    A discord.py cog with commands pertaining to the a Text Filter for admins to monitor their server
+    """
+
+    def __init__(self, bot, database_manager=None):
+        if not database_manager:
+            database_manager = KoalaBot.database_manager
+        self.bot = bot
+        database_manager.create_base_tables()
+        database_manager.insert_extension("TextFilter", 0, True, True)
+        self.tf_database_manager = TextFilterDBManager(database_manager, bot)
+        self.tf_database_manager.create_tables()
+
+    @commands.command(name="filter", aliases=["filter_word"])
+    @commands.check(KoalaBot.is_admin)
+    @commands.check(text_filter_is_enabled)
+    async def filter_new_word(self, ctx, word, filter_type="banned", too_many_arguments=None):
+        """
+        Adds a new word to the filtered text list
+        :param ctx: The discord context
+        :param word: The first argument and word to be filtered
+        :param filter_type: The filter type (banned or risky)
+        :param too_many_arguments: Used to check if too many arguments have been given
+        :return:
+        """
+        error = """Something has gone wrong, your word may already be filtered or you have entered the 
+                command incorrectly. Try again with: `k!filter [filtered_text] [[risky] or [banned]]`"""
+        if too_many_arguments is None and type_exists(filter_type):
+            await filter_text(self, ctx, word, filter_type, False)
+            await ctx.channel.send("*" + word + "* has been filtered as **"+filter_type+"**.")
+            return
+        raise Exception(error)
```

Comment:
In future, it is best to use specific exceptions, or create your own, not just the base one. It's ok for this cog though, just something to keep in mind in future